src/Application/Digital/Commands/RestoreEntry.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Application.Common.Models;
+using Application.Common.Models.Dtos.Digital;
+using AutoMapper;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace Application.Digital.Commands;
+
+public class RestoreEntry
+{
+    public record Command : IRequest<EntryDto>
+    {
+        public User CurrentUser { get; init; } = null!;
+        public Guid EntryId { get; init; }
+    }
+
+    public class Handler : IRequestHandler<Command, EntryDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IDateTimeProvider _dateTimeProvider;
+        private const string BinString = "_bin";
+        
+        public Handler(IApplicationDbContext context, IMapper mapper, IDateTimeProvider dateTimeProvider)
+        {
+            _context = context;
+            _mapper = mapper;
+            _dateTimeProvider = dateTimeProvider;
+        }
+
+        public async Task<EntryDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var entry = await _context.Entries.FirstOrDefaultAsync(x => x.Id == request.EntryId, cancellationToken);
+
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist");
+            }
+
+            var path = entry.Path;
+            var firstSlashIndex = path.IndexOf("/", StringComparison.Ordinal);
+            var binCheck = path.Substring(0, firstSlashIndex);
+
+            if (!binCheck.Contains(BinString))
+            {
+                throw new ConflictException("Entry is not in bin");
```

Comment:
By idempotency standards, this should not be thrown. A 204 No content should be returned instead